Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp

```diff
@@ -83,90 +237,97 @@ namespace AZ::Internal
             {
                 AZ::IO::FixedMaxPath m_path;
                 FixedValueString m_moniker;
+                AZ::SemanticVersion m_version;
             };
 
-            struct EnginePathsVisitor : public AZ::SettingsRegistryInterface::Visitor
+            AZStd::set<AZ::IO::FixedMaxPath> searchedProjectPaths;
+            AZStd::vector<EngineInfo> searchedEngineInfo;
+
+            if (manifestLoaded)
             {
-                using AZ::SettingsRegistryInterface::Visitor::Visit;
-                void Visit(
-                    const AZ::SettingsRegistryInterface::VisitArgs& visitArgs, AZStd::string_view value) override
-                {
-                    m_enginePaths.emplace_back(EngineInfo{ AZ::IO::FixedMaxPath{value}.LexicallyNormal(), FixedValueString{visitArgs.m_fieldName} });
-                    // Make sure any engine paths read from the manifest are absolute
-                    AZ::IO::FixedMaxPath& recentEnginePath = m_enginePaths.back().m_path;
-                    if (recentEnginePath.IsRelative())
+                const auto engineVersionKey = FixedValueString::format("%s/version", EngineSettingsRootKey);
+                const auto enginesKey = FixedValueString::format("%s/engines", O3deManifestSettingsRootKey);
+
+                AZ::IO::FixedMaxPath projectUserPath;
+                settingsRegistry.Get(projectUserPath.Native(), FilePathKey_ProjectUserPath);
+
+                EngineInfo mostCompatibleEngineInfo;
+
+                // Look through the manifest engines for the most compatible engine
+                AZ::SettingsRegistryVisitorUtils::VisitArray(settingsRegistry, [&](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
```

Comment:
FYI the `visitArgs` has an `m_registry` member with a reference to the SettingsRegistry on it